src/main/java/org/commcare/formplayer/objects/SerializableMenuSession.java

```diff
@@ -1,30 +1,61 @@
-package org.commcare.formplayer.repo;
+package org.commcare.formplayer.objects;
 
 import org.commcare.formplayer.session.MenuSession;
+import org.hibernate.annotations.GenericGenerator;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import javax.persistence.*;
 
 /**
  * Created by willpride on 8/1/16.
  */
+
+@Entity
+@Table(name="menu_sessions")
+@EntityListeners(AuditingEntityListener.class)
 public class SerializableMenuSession {
 
+    @Id @GeneratedValue(generator="system-uuid")
+    @GenericGenerator(name="system-uuid", strategy = "uuid")
     private String uuid;
+
+    @Column(name="username")
```

Comment:
these annotations are only needed if you want to change the defaults. The currently configured naming strategy will:
- Replace dots with underscores
- Change camel case to snake case, and
- Lower-case table names

This doesn't always match up with what the table / column names are in the DB. If that's the case then we need to use the annotation to customize it. For columns like this `username` there are no changes from the default so we can leave it off altogether.

Reply:
Thanks for the very detailed note, Simon!